api/envoy/extensions/network/dns_resolver/apple/v3/appl_dns_resolver.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package envoy.extensions.network.dns_resolver.apple.v3;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.network.dns_resolver.apple.v3";
+option java_outer_classname = "ApplDnsResolverProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: apple DNS resolver configuration]
+
+// Configuration for apple DNS resolver
+// For Envoy running on Apple system, to use apple DNS library as DNS resolver, the following need to be done:
+// 1) In file: source/extensions/extensions_build_config.bzl, enable Apple DNS extension by adding below line
+//      "envoy.dns_resolver.apple":                "//source/extensions/network/dns_resolver/apple:apple_dns_lib",
```

Comment:
Do we want to do this by default with some Bazel magic on Apple platforms? This would avoid making things harder than necessary for existing users to migrate.

Reply:
Yes,   MacOS build with Apple DNS extension enabled, if one has the run time flag: `envoy.restart_features.use_apple_api_for_dns_lookups`` enabled, Envoy will use  Apple DNS library even one doesn't have the apple typed_dns_resolver_config in place. Please check the latest implementation in dns_factory.h:makeDnsResolverConfig.  Also PTAL the latest comments in this file.